lib/features/base/mixin/save_media_to_gallery_mixin.dart

```diff
@@ -0,0 +1,137 @@
+  Future<void> handleAndroidStoragePermission(BuildContext context) async {
+    if (await StoragePermissionService().isUserHaveToRequestStoragePermissionAndroid()) {
+      final permission = await Permission.storage.request();
+
+      if (permission.isPermanentlyDenied && context.mounted) {
+        showDialog(
+          useRootNavigator: false,
+          context: context,
+          builder: (_) {
+            return PermissionDialog(
+              icon: const Icon(Icons.storage_rounded),
+              permission: Permission.storage,
+              explainTextRequestPermission: Text(
+                AppLocalizations.of(context).explainPermissionToDownloadFiles(
+                  AppLocalizations.of(context).app_name,
+                ),
+              ),
+              onAcceptButton: () =>
+                  StoragePermissionService().goToSettingsForPermissionActions(),
+            );
+          },
+        );
+      }
+
+      if (!permission.isGranted) {
+        log('SaveMediaToGalleryMixin::handleAndroidStoragePermission: Permission Denied');
+        throw StoragePermissionException("Don't have permission to save file");
+      }
+    }
+  }
+
+  Future<void> handlePhotoPermissionIOS(BuildContext context) async {
+    final permissionStatus = await StoragePermissionService().requestPhotoAddOnlyPermissionIOS();
+    if (permissionStatus.isPermanentlyDenied && context.mounted) {
+      showDialog(
+        useRootNavigator: false,
+        context: context,
+        builder: (_) {
+          return PermissionDialog(
+            icon: const Icon(Icons.photo),
+            permission: Permission.photos,
+            explainTextRequestPermission: Text(
+              AppLocalizations.of(context).explainPermissionToGallery(
+                AppLocalizations.of(context).app_name,
+              ),
+            ),
+            onAcceptButton: () =>
+                StoragePermissionService().goToSettingsForPermissionActions(),
+          );
+        },
+      );
+    }
+    if (!permissionStatus.isGranted) {
+      throw StoragePermissionException('Permission denied');
+    }
+  }
+
+  Future<void> saveMediaToGallery({
+    required File fileInDownloadsInApp,
+    required MediaType mediaType,
+    OnSaveCallbackAction? onSaveCallbackAction
+  }) async {
+    if (mediaType.isImageFile()) {
+      await saveImageToGallery(file: fileInDownloadsInApp);
+    } else if (mediaType.isVideoFile()) {
+      await saveVideoToGallery(file: fileInDownloadsInApp);
+    } else {
+      return;
+    }
+    onSaveCallbackAction?.call(true);
+  }
+
+  Future<void> saveImageToGallery({
+    required File file,
+  }) async {
+   log('SaveMediaToGalleryMixin::saveImageToGallery:file path: ${file.path}');
+    await Gal.putImage(file.path);
+  }
+
+  Future<void> saveVideoToGallery({
+    required File file,
+  }) async {
+    log('SaveMediaToGalleryMixin::saveVideoToGallery:file path: ${file.path}');
+    await Gal.putVideo(file.path);
+  }
```

Comment:
Consider making these methods private. We should only expose methods that are going to be called publicly.

Reply:
done